### PR TITLE
[SYCL] tests for ONEAPI_DEVICE_SELECTOR.

### PR DESCRIPTION
**Pull request by cperkinsintel**
ONEAPI_DEVICE_SELECTOR is being implemented here: https://github.com/intel/llvm/pull/6779/

These are its tests.

### CODE REVIEW

File: SYCL/OneapiDeviceSelector/cpu_gpu_acc.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// REQUIRES: cpu, gpu, accelerator, opencl
+
+// RUN: %clangxx -fsycl %s -o %t.out
+
+// RUN: env ONEAPI_DEVICE_SELECTOR=opencl:acc %t.out | FileCheck %s --check-prefixes=CHECK-ACC-ONLY
```
Comment (gmlueck) [inline SYCL/OneapiDeviceSelector/cpu_gpu_acc.cpp]: There is no `acc` in the grammar.

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by cperkinsintel** [comment]
As these tests depend upon https://github.com/intel/llvm/pull/6779 the "normal" runs will fail. The run named "Jenkins/llvm-test-suite" is the run against the PR that introduces the selectors and that run is passing.

**Pull request merged**